numpy/ma/core.py

```diff
@@ -6535,9 +6530,7 @@ def compressed(x):
         Equivalent method.
 
     """
-    if not isinstance(x, MaskedArray):
-        x = asanyarray(x)
-    return x.compressed()
+    return asanyarray(x).compressed()
```

Comment:
I think with your other change it is now safe (and shorter though a bit slower) to just do 
```
compressed = _frommethod('compressed')
```